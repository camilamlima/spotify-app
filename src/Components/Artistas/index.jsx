import React, {Component} from 'react'
import Card from '../Card';
import Spotify from 'spotify-web-api-js';

class Artistas extends Component{
    
    constructor(props) {
        super(props);

        this.spotifyApi = new Spotify()
        this.spotifyApi.setAccessToken(localStorage.getItem("token"));
        
        this.state = {
          artistas: []
        };
     }
     componentDidMount (){
       this.search();
     }
  
    search(){
        console.log("Buscando na API")
        let artistas = ["Alcymar Monteiro", "Leela", "Ney Alves"]
        artistas.map((artista) => {
          this.spotifyApi.searchArtists(artista)
            .then(function(data) {
              
              let item = data.artists.items[0];
              let s_artistas = this.state.artistas;
              
              s_artistas.push({
                  name: item.name,
                  href: item.external_urls.spotify,
                  image: item.images[0].url
              })
              
              this.setState({artistas: s_artistas})
              console.log(item);
            }.bind(this), function(err) {
              console.error(err);
            });
          
          
          
        });
        
    }
  
    render(){
        return(
          <div className="jumbotron jumbotron-devices">
        
            <div className="container">
              <div className="row">
                {this.state.artistas.map((item, i) => {
                  return (
                    <div className="col-lg-4" key={i}>
                      <Card artista={item} />
                    </div>
                  );
                })}
              </div>
            </div>
          </div>
        );
  }
}    
                    
export default Artistas;