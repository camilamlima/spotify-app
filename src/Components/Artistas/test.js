import React from 'react';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Artistas from '.';

test('Testar componente de Artistas da aplicação', () => {

    const component = renderer.create(
      <BrowserRouter>
        <Artistas />
      </BrowserRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});