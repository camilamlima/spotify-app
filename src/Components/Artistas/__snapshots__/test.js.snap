// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testar componente de Artistas da aplicação 1`] = `
<div
  className="jumbotron jumbotron-devices"
>
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
